### PR TITLE
Added new tests for experiment without Try

### PR DESCRIPTION
**Pull request by arntj**
Added new test to ensure that experiment returns correctly even without any candidate. Also added tests to verify that the Clean, RunIf and Enabled delegates are not run and no publish is done if there are no candidates.

### CODE REVIEW

File: test/Scientist.Test/ScientistTests.cs [hunk 1]
```diff
@@ -744,5 +744,84 @@ public void KeepingItClean()
...
+        public void NoCandidateMeansNoPublish()
+        {
+            const string experimentName = "noCandidateNoPublish";
+
+            var result = Scientist.Science<int>(experimentName, experiment =>
+            {
+                experiment.Use(() => 42);
+            });
+
+            Assert.False(TestHelper.Results<int>().Any(m => m.ExperimentName == experimentName));
```
Comment (joncloud) [inline test/Scientist.Test/ScientistTests.cs]: Consider adding a message to this method call to provide some feedback to the test runner.

Comment (arntj) [inline test/Scientist.Test/ScientistTests.cs]: @joncloud Thanks for your suggestion. I've added a message.

File: test/Scientist.Test/ScientistTests.cs [hunk 2]
```diff
@@ -744,5 +744,85 @@ public void KeepingItClean()
 
             mock.Received().Clean(expectedResult);
         }
+
+        [Fact]
+        public void NoCandidate()
```
Comment (Haacked) [inline test/Scientist.Test/ScientistTests.cs]: I think the method name should communicate the expected outcome. For example, `NoCandidateStillRunsControl`.

**Comment by Haacked** [comment]
Looks good! One minor nit about a test method name. Also, I merged some changes in master that conflict with this. You'll either need to rebase this against master and force push to your branch OR just merge upstream master in your branch.

**Comment by arntj** [comment]
@Haacked Thanks for your suggestions. I've merged the latest changes and changed the method name as you suggested. :thumbsup:

**Comment by Haacked** [comment]
Thanks!

selfie-0

**Pull request merged**